redhat-appstudio/infra-deployments
# refactor pipelines

This PR contains the following changes. 

- move pipelines to OCI build-definitions repository which contains container builds for pipelines and task containers.
- see `https://github.com/redhat-appstudio/build-definitions/tree/main/pipelines`
- add CM in build-templates to point to OCI bundle
- in devmode enable override for pipelines (devmode)
- refactor cluster tasks to correct location in this repo
   see    `https://github.com/redhat-appstudio/build-definitions/tree/main/tasks`
- additional helper cluster tasks
- enable OCI buildles via feature flags on openshift-pipelines 
How to test:
1. Checkout the PR into a branch from main  
2. Push that branch to git, no changes yet to the next step has a branch to sync from 
3. run `./hack/development.sh` 
4. commit the overlay changes, and push the new branch to git. 
5. create a new project `oc new-project demo`
5. Option 1 - run `./hack/build/m2-builds.sh` and all builds will run in parallel 
6. Option 2 -run `./hack/build/m2-builds.sh -deploy` to test builds and deploys one at a time

In option 1, just the builds will be performed
In option 2, the builds will also be deployed 

In the developer view, you will see the deployed applications appear one at a time as they complete the builds.
